[DOC-FIX] Cannot run pytests due to missing dependencies in default development environment

### Willingness to contribute

No. I cannot contribute a documentation fix at this time.

### URL(s) with the issue

CONTRIBUTING.rst

### Description of proposal (what needs changing)

I followed the `CONTRIBUTING.rst` and ran `dev/dev-env-setup.sh -d .venvs/mlflow-dev -q` in a fresh clone of the mlflow repo to create an virtual environment to develop within. However, when I ran `./dev/run-python-tests.sh` to run the tests, I got the error
```
ImportError while loading conftest '/Users/jerry.liang/mlflow/tests/conftest.py'.
tests/__init__.py:1: in <module>
    from mlflow.utils.logging_utils import _configure_mlflow_loggers
mlflow/__init__.py:32: in <module>
    import mlflow.tracking._model_registry.fluent
mlflow/tracking/__init__.py:8: in <module>
    from mlflow.tracking.client import MlflowClient
mlflow/tracking/client.py:13: in <module>
    import yaml
E   ModuleNotFoundError: No module named 'yaml'
```
After installing `pyyaml` and rerunning the command, I got the error `ModuleNotFoundError: No module named 'google'`. I'm assuming that the environment setup script doesn't install all the test requirements?

--- a/dev/dev-env-setup.sh
+++ b/dev/dev-env-setup.sh
@@ -220,11 +220,12 @@ echo "$(tput setaf 3)Activated environment is located: $(tput bold) $directory/b
 
 echo "Installing pip dependencies for development environment."
 
+# Install current checked out version of MLflow (local)
+pip install $(quiet_command) -e .[extras]
+
 if [[ -n "$full" ]]; then
   # Install dev requirements and test plugin
   pip install $(quiet_command) -r "$MLFLOW_HOME/requirements/dev-requirements.txt"
-  # Install current checked out version of MLflow (local)
-  pip install $(quiet_command) -e .[extras]
   # Install test plugin
   pip install $(quiet_command) -e "$MLFLOW_HOME/tests/resources//mlflow-test-plugin"
   echo "Finished installing pip dependencies."

